### PR TITLE
Fix serverTrust crash

### PR DESCRIPTION
**Pull request by Kaspik**
### Issue Link :link:
- Fixes https://github.com/AFNetworking/AFNetworking/issues/4541

### Goals :soccer:
- Release of the new version 4.0.1 with no crash.

### Implementation Details :construction:
`challenge.protectionSpace.serverTrust` is nil if the authentication method of the protection space is not server trust

### CODE REVIEW

**Comment by 0xced** [comment]
Thanks for the fix! Can you add a unit test that triggers the crash (without your fix) in order to avoid a future potential regression?

**Comment by 0xced** [comment]
@Kaspik May I ask why did you close this pull request?

**Comment by Kaspik** [comment]
Hah, I merged my other PR and it closed this one. I'll reopen.

**Comment by 0xced** [comment]
Are you comfortable to add a unit test that triggers the crash (without your fix) in order to avoid a future potential regression or would you prefer me to add the unit test?

**Comment by Kaspik** [comment]
I'm already out from my computer for the weekend, so feel free to do it otherwise I'll try on Monday. :)

**Comment by 3a4oT** [comment]
ping @Kaspik

**Comment by Kaspik** [comment]
As `static NSError * ServerTrustError(SecTrustRef serverTrust, NSURL *url)` was private, I ended up updating it to func that is publicly exposed, then adding test it throws if serverTrust is nil.

Let me know if you have any other idea how to revert this but still test it.

File: AFNetworking/AFURLSessionManager.h [hunk 1]
```diff
@@ -447,6 +447,8 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)setDownloadTaskDidResumeBlock:(nullable void (^)(NSURLSession *session, NSURLSessionDownloadTask *downloadTask, int64_t fileOffset, int64_t expectedTotalBytes))block;
 
++ (nonnull NSError *)serverTrustErrorWithServerTrust:(nonnull SecTrustRef)serverTrust url:(nonnull NSURL *)url;
```
Comment (jshier) [inline AFNetworking/AFURLSessionManager.h:450]: I'd rather not make this method public. Instead of testing the error method directly, perhaps you can test the delegate method which creates the error? I do like the nullability annotations though, so applying those to the function might be helpful.

Comment (Kaspik) [reply AFNetworking/AFURLSessionManager.h:450]: I'm not sure how to "repro" the case where it returns with no server trust - so if anyone has, feel free to give me a hint or final solution.

**Comment by jshier** [comment]
Merging to branch to add tests but will credit this PR with the change.
